tests/testing_tests/test_causal_test_suite.py

```diff
@@ -0,0 +1,132 @@
+import unittest
+import os
+import numpy as np
+import pandas as pd
+from causal_testing.testing.causal_test_engine import CausalTestEngine
+from causal_testing.testing.causal_test_engine import CausalTestSuite
+from causal_testing.testing.causal_test_case import CausalTestCase
+from causal_testing.testing.base_test_case import BaseTestCase
+from causal_testing.specification.variable import Input, Output
+from causal_testing.testing.causal_test_outcome import ExactValue
+from causal_testing.testing.estimators import CausalForestEstimator, LinearRegressionEstimator
+from causal_testing.specification.causal_specification import CausalSpecification, Scenario
+from causal_testing.data_collection.data_collector import ObservationalDataCollector
+from tests.test_helpers import create_temp_dir_if_non_existent, remove_temp_dir_if_existent
+from causal_testing.specification.causal_dag import CausalDAG
+
+
+class TestCausalTestSuite(unittest.TestCase):
+    """Test the Test Suite object and it's implementation in the test engine using dummy data."""
+
+    def setUp(self) -> None:
+        # 1. Create dummy Scenario and BaseTestCase
+        A = Input("A", float)
+        self.A = A
+        C = Output("C", float)
+        self.C = C
+        D = Output("D", float)
+        self.D = D
+        self.base_test_case = BaseTestCase(A, C)
+        self.scenario = Scenario({A, C, D})
+
+        # 2. Create DAG and dummy data and write to csvs
+        temp_dir_path = create_temp_dir_if_non_existent()
+        dag_dot_path = os.path.join(temp_dir_path, "dag.dot")
+        dag_dot = """digraph G { A -> C; D -> A; D -> C}"""
+        f = open(dag_dot_path, "w")
+        f.write(dag_dot)
+        f.close()
+
+        np.random.seed(1)
+        df = pd.DataFrame({"D": list(np.random.normal(60, 10, 1000))})  # D = exogenous
+        df["A"] = [1 if d > 50 else 0 for d in df["D"]]
+        df["C"] = df["D"] + (4 * (df["A"] + 2))  # C = (4*(A+2)) + D
+        self.observational_data_csv_path = os.path.join(temp_dir_path, "observational_data.csv")
+        df.to_csv(self.observational_data_csv_path, index=False)
```

Comment:
This is standard across our tests, to create test data dynamically, rather than cache it somewhere?

Reply:
It is how the current `causal_test_engine` tests create / use data:

https://github.com/CITCOM-project/CausalTestingFramework/blob/main/tests/testing_tests/test_causal_test_engine.py.

I've not played with caching test data, but it makes sense and I'll make an issue to be looked into